tox-offline-draft.md

```diff
@@ -5,77 +5,154 @@
 further work and development of the Tox protocol.
 
+## Terminology
+PoW:
+  Proof-of-Work -- Determines CPU work necessary in order to store an offline message on the DHT.
+
+SENDER:
+  The friend of RECEIVER sending an offline message to her when she is offline.
+
+RECEIVER:
+  The friend of SENDER receiving an offline message.
+
+STORAGE_NODE:
+  One of potentially many nodes storing offline messages for other users.
+
+OM Keys:
+  Offline Messaging public / private keypair.
+
 ## DHT Extension
 
-The DHT must be extended in order to be able to store arbitrary data. Tox
-Clients should have a reasonable default for DHT storage size set and
+The DHT must be extended in order to be able to store offline messages. Tox
+Clients should have a reasonable default set for DHT storage size and
 optionally allow users to change it. This space will be used for storing
 messages of other users.
 
-## Inserting Messages 
+## Storage Space Allocation
 
-Analagous to dpush [0], a user creates a new public and private key pair
-(Current DHT keys are ephemeral) to sign their inbox location. The inbox is
-stored in the DHT on computers who are in charge of that key range. (TODO:
-Should the public key be hashed?) Users then store messages they want to
-send to a store in that inbox's location. 
+The RECEIVER must request a certain amount of storage space for incoming
+offline messages on a number of STORAGE_NODEs. A PoW job must be completed
+by the RECEIVER to do this. The more space required per node, the higher
+the PoW difficulty.
 
-(TODO: Inbox location can be changed if necessary, as long as new location
-is always signed by OM key. Is this a useful feature? If you change you
-lose all old messages stored in old locations).
+## Inserting Messages
 
-## Retrieving Messages
+A user creates a new public and private key pair (Current DHT keys are
+expected to be ephemeral) per friend to encrypt offline messages.  Messages are stored
+amongst an array of known nodes.  Users then store messages they want to send
+to a store on those nodes after optionally completing a PoW function. 
 
-Users simply check their inbox location in the DHT when they come online.
-If any messages are available they will download them from the nodes
-holding them and decrypt them with their OM private keys. Nodes holding
-messages should them delete them from their store.
+## Retrieving Messages
 
-(TODO: Similar to what dpush does, we could have the receiver scan sender's
-public keyspaces for messages intended for them, thus requiring no PoW from
-the sender. I feel like this can reintroduce spam though, however the
-receiver is fully in control of what they want to pull at this point.)
+Users simply check the SENDER's STORAGE NODES in the DHT when they come
+online.  If any messages are available they will download them from the nodes
+holding them and decrypt them with their OM keys. Nodes holding messages
+should then delete them from their store.
```

Comment:
Will multiple nodes hold single message? If RECEIVER will get message from single one, how all others would know that it should be deleted?